core/src/callback.jl

```diff
@@ -141,6 +193,99 @@ function update_cumulative_flows!(u, t, integrator)::Nothing
             end
         end
     end
+
```

Comment:
I could use some some inline documentation on the order of computations and why you need it. Also maybe in the docs somewhere?